### DISCUSSION TITLE
Save load functionality

### DISCUSSION THREAD

**Discussion by oysteinsolheim** [JuliaReinforcementLearning/ReinforcementLearning.jl#183] [Q&A]
As far as I understand the save/load functionality of agents/policies has now been removed from this package. But What is then the preferred way to do it? I think I read a discussion about it once but cannot remember where or when. Is the solution to use Flux-functionality for loading and saving network parameters?

Rgds,
Øystein

**Comment by findmyway** [comment]
Hello Øystein,

Nice to see you again here🤗

Yes you are right that load/save related functions are removed. The idea behind the decision is to let users to decide which kind of tool to use.

Following are my suggestions:

1. If you want to save a policy and make it compatiable accross different versions of ReinforcementLearning.jl, you can use Flux.trainable to get the parameters and save it with BSON.jl. This is the most safe one unless the specific model structure changed across different versions.
2. If you don't care about compatibility, then simply use any tool you like (just remember to move data from gpu to cpu first ) BSON.jl or JLD.jl JLD2.jl JLSO.jl
3. If you want to save the trajectory generated during training, then BSON.jl and JLSO.jl are not recommended. The former has a bug for large data and the later is too slow(but since it supports compression, you may still choose it at some point ). JLD.jl is the fastest one but I heard it had some unacceptable bugs. Not sure if they still exist.
